File: core/ErChaSouSuoShuDeDiKdaJieDianLcof/ErChaSouSuoShuDeDiKdaJieDianLcof.java
package com.dbc;

import java.util.Stack;

public class ErChaSouSuoShuDeDiKdaJieDianLcof {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    public int kthLargest(TreeNode root, int k) {
        Stack<TreeNode> stack = new Stack<>();
        TreeNode cur = root;
        int count = 0;
        while (!stack.isEmpty() || cur != null) {
            while (cur != null) {
                stack.add(cur);
                cur = cur.right;
            }
            cur = stack.pop();
            count++;
            if (count == k) return cur.val;
            cur = cur.left;
        }
        return -1;
    }
}

File: core/ErChaSouSuoShuDeDiKdaJieDianLcof/ErChaSouSuoShuDeDiKdaJieDianLcof.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


def kthLargest(root: TreeNode, k: int) -> int:
    stack, count, cur = [], 0, root
    while stack or cur:
        while cur:
            stack.append(cur)
            cur = cur.right
        cur = stack.pop()
        count += 1
        if count == k:
            return cur.val
        cur = cur.left

    return -1

